### PR TITLE
Added min and max functions for Vector3

### PR DESCRIPTION
**Pull request by Frago9876543210**

### CODE REVIEW

File: src/Vector3.php [hunk 1]
```diff
@@ -273,6 +273,14 @@ public function distance(Vector3 $pos) : float{
 	public function distanceSquared(Vector3 $pos) : float{
 		return (($this->x - $pos->x) ** 2) + (($this->y - $pos->y) ** 2) + (($this->z - $pos->z) ** 2);
 	}
+	
+	public function min(Vector3 $v) : Vector3{
+		return new Vector3(min($this->x, $v->x), min($this->y, $v->y), min($this->z, $v->z));
```
Comment (dktapps) [inline src/Vector3.php]: Please add a `use function min;` to the import section.

**Comment by SOF3** [comment]
+1 static method. Although I'd rather see PHP to add static methods automatically for all instance methods :P

**Comment by dktapps** [comment]
Please change these methods to be static. They don't belong to the context of either vector.

**Comment by dktapps** [comment]
There are other methods in here which should probably be static-ified too, but that would involve BC breaks.

File: src/Vector3.php [hunk 2]
```diff
@@ -273,6 +274,14 @@ public function distance(Vector3 $pos) : float{
 	public function distanceSquared(Vector3 $pos) : float{
 		return (($this->x - $pos->x) ** 2) + (($this->y - $pos->y) ** 2) + (($this->z - $pos->z) ** 2);
 	}
+	
+	public static function min(Vector3 $v) : Vector3{
+		return new Vector3(min($this->x, $v->x), min($this->y, $v->y), min($this->z, $v->z));
+	}
+
+	public static function max(Vector3 $v) : Vector3{
+		return new Vector3(max($this->x, $v->x), max($this->y, $v->y), max($this->z, $v->z));
```
Comment (dktapps) [inline src/Vector3.php]: `$this` in static context?

File: src/Vector3.php [hunk 3]
```diff
@@ -273,6 +274,14 @@ public function distance(Vector3 $pos) : float{
 	public function distanceSquared(Vector3 $pos) : float{
 		return (($this->x - $pos->x) ** 2) + (($this->y - $pos->y) ** 2) + (($this->z - $pos->z) ** 2);
 	}
+	
+	public static function min(Vector3 $v) : Vector3{
+		return new Vector3(min($this->x, $v->x), min($this->y, $v->y), min($this->z, $v->z));
```
Comment (dktapps) [inline src/Vector3.php]: `$this` in static context?

**Pull request closed without merging**